### ISSUE TITLE
Failing to parse specific json

### ISSUE THREAD

**Issue by ZenVoich** [NatLabs/serde#11]
serde 1.0.2 
moc 0.9.8

```motoko diff
let jsonText = "[{ \"a\": [] }, { \"a\": [ \"x\" ] }]";
type Res = [{ a : [Text] }];
let res : ?Res = from_candid(JSON.fromText(jsonText, null));
```

Failing with error: ```wasm trap: wasm `unreachable` instruction executed```

If I add item to first empty array - parses successfully:
```diff
-let jsonText = "[{ \"a\": [] }, { \"a\": [ \"x\" ] }]";
+let jsonText = "[{ \"a\": [\"v\"] }, { \"a\": [ \"x\" ] }]";
type Res = [{ a : [Text] }];
let res : ?Res = from_candid(JSON.fromText(jsonText, null));
```

**Comment by tomijaga** [comment]
Ahh, I see what is going on here. 
I set the default type for the empty arrays to `#vector(#empty))` so the first record's type conflicts with the second record, which is `#vector(#text))`. 
I'll need to add functionality to scan the items in the array to find the array's intended type.

I'll update you once I've made these changes.

**Comment by tomijaga** [comment]
Hey @ZenVoich,
The issue should be fixed now in `serde@2.0.0`.

Please let me know if there is anything else I can help with or improve in the library.
Cheers!

**Comment by ZenVoich** [comment]
@tomijaga Confirmed, now parses without errors. Thank you!
